.github/workflows/ci.yaml

```diff
@@ -37,9 +37,60 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
-      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+      # - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491
+      #   with:
+      #     go-version-file: 'go.mod'
+      #     cache: false
+      - name: Check out platform
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
         with:
-          go-version-file: 'go.mod'
-          cache: false
-      - name: e2e Tests
-        run: echo "No e2e tests yet." && exit 0
+          repository: opentdf/platform
+          path: platform
+      - name: Set up go
+        uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7
+        with:
+          go-version: "1.22.3"
+          check-latest: false
+          cache-dependency-path: |
+            platform/service/go.sum
+            platform/examples/go.sum
+            platform/protocol/go/go.sum
+            platform/sdk/go.sum
+      - run: |
+          .github/scripts/init-temp-keys.sh
+          cp opentdf-dev.yaml opentdf.yaml
+        working-directory: platform
+      - name: Added Trusted Certs
+        run: |
+          sudo chmod -R 777 ./keys
+          sudo apt-get install -y ca-certificates
+          sudo cp ./keys/localhost.crt /usr/local/share/ca-certificates
+          sudo update-ca-certificates
+        working-directory: platform
+      - run: docker compose up -d --wait --wait-timeout 240
+        working-directory: platform
+      - run: go run ./service provision keycloak
+        working-directory: platform
+      - run: go run ./service provision fixtures
+        working-directory: platform
+      - uses: JarvusInnovations/background-action@2428e7b970a846423095c79d43f759abf979a635
+        name: start server in background
+        with:
+          run: >
+            go build -o opentdf -v service/main.go
+            && .github/scripts/watch.sh opentdf.yaml ./opentdf start
+          wait-on: |
+            tcp:localhost:8080
+          log-output-if: true
+          wait-for: 90s
+          working-directory: platform
+
+      - name: build the CLI
+        run: go build .
+      - name: set up the config
+        run: cp otdfctl-example.yaml otdfctl.yaml
+      - name: run encrypt -> decrypt
+        run: |-
```

Comment:
I don't have a lot to add yet but this looks good. I wonder if we should do something like the below to get it a little drier as we extend this:

```shell
echo -n '{"clientId":"opentdf","clientSecret":"secret"}' > creds.json
export WITH_CREDS='--with-client-creds-file ./creds.json'

# then we can use
./otdfctl <something> $WITH_CREDS --some-flag
```